#!/bin/sh /usr/share/dpatch/dpatch-run
# Description: Apply changeset 27248 for kernels with 2.6.33 drm implementation
# Author: Michael Meskes <meskes@debian.org>

@DPATCH@
diff -urNad virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/linux/drm/vboxvideo_drm.c.orig virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/linux/drm/vboxvideo_drm.c
--- virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/linux/drm/vboxvideo_drm.c.orig	2010-03-23 15:37:10.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/linux/drm/vboxvideo_drm.c	2010-03-23 15:44:57.000000000 +0100
@@ -87,7 +87,13 @@
 		 .owner = THIS_MODULE,
 		 .open = drm_open,
 		 .release = drm_release,
+                 /* This was changed with Linux 2.6.33 but Fedora backported this 
+                  * change to their 2.6.32 kernel. */ 
+#if defined(DRM_UNLOCKED) || LINUX_VERSION_CODE >= KERNEL_VERSION (2, 6, 33) 
+                 .unlocked_ioctl = drm_ioctl, 
+#else 
 		 .ioctl = drm_ioctl,
+#endif
 		 .mmap = drm_mmap,
 		 .poll = drm_poll,
 		 .fasync = drm_fasync,
